### PR TITLE
lp1733215: Mapping for  DENON MC6000MK2

### PR DESCRIPTION
**Pull request by uklotzde**
Not yet finished, I just copied the files from my resources folder. Basically it should work and the effect sections (metaknobs, enable/disable buttons) are mapped.

I will get physical access to the controller in a few days and an update will follow. Until then this should be a usable starting point for 2.1. It still contains a lot of legacy cruft that need to be replaced by the new components.

### CODE REVIEW

**Comment by Be-ing** [comment]
Leaving legacy cruft in here is okay as long as it works for the release.

**Comment by Be-ing** [comment]
I would prefer to have _everything_ including controller mappings done for the release candidate on Tuesday April 3, but I think we can make an exception for this considering you won't have access to the hardware before that.

File: res/controllers/Denon-MC6000MK2-scripts.js [hunk 1]
```diff
@@ -1212,22 +1162,28 @@ DenonMC6000MK2.getEfxUnitByGroup = function(group) {
...
+    this.paramGroups = [];
     this.params = [];
-    this.params[1] = new DenonMC6000MK2.EfxParam(this);
-    this.params[2] = new DenonMC6000MK2.EfxParam(this);
-    this.params[3] = new DenonMC6000MK2.EfxParam(this, this.group, "mix");
-    this.dryWetMixParam = this.params[3];
-    this.dryWetMixParam.adjustParam(0.5); // mix (50% dry / 50% wet)
+    this.paramGroups[1] = "[" + DenonMC6000MK2.EFX_RACK + "_" + unit + "_Effect1]";
+    DenonMC6000MK2.efxUnitsByGroup[this.paramGroups[1]] = this;
+    this.params[1] = new DenonMC6000MK2.EfxParam(this, this.paramGroups[1], "meta");
```
Comment (Be-ing) [inline res/controllers/Denon-MC6000MK2-scripts.js]: Use the [EffectUnit object from the Components library](https://mixxx.org/wiki/doku.php/components_js#effectunit). It behaves the same way by default and allows the user to control the parameters of each effect.

**Comment by uklotzde** [comment]
Especially the effects mapping was a quick hack on top of an unfinished _components_ migration just before an event. I hope to get rid of the manual side/deck management eventually, but recently I didn't have enough time to even start working on it again.

**Pull request merged**